app/screens/settings/display_theme/index.ts

```diff
@@ -0,0 +1,61 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
+import withObservables from '@nozbe/with-observables';
+import {combineLatest, of as of$} from 'rxjs';
+import {switchMap} from 'rxjs/operators';
+
+import {Preferences} from '@constants';
+import {queryPreferencesByCategoryAndName} from '@queries/servers/preference';
+import {
+    observeAllowedThemesKeys,
+    observeConfigBooleanValue,
+    observeCurrentTeamId,
+    observeCurrentUserId,
+} from '@queries/servers/system';
+import {WithDatabaseArgs} from '@typings/database/database';
+import {safeParseJSON} from '@utils/helpers';
+
+import DisplayTheme from './display_theme';
+
+const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
+    const currentTeamId = observeCurrentTeamId(database);
+    const currentUserId = observeCurrentUserId(database);
+    const themePreferences = queryPreferencesByCategoryAndName(database, Preferences.CATEGORY_THEME).observe();
+    const configAllowCustomThemes = observeConfigBooleanValue(database, 'AllowCustomThemes');
+
+    const customTheme = combineLatest([currentTeamId, themePreferences, configAllowCustomThemes]).pipe(
```

Comment:
actually, I would simply this with the use of `useTheme` and inspecting if the type if `custom`, then for the ability to roll it back I would store that theme value in a local `customTheme` state

Reply:
🤯